src/udp_wrap.cc

```diff
@@ -233,16 +233,16 @@ void UDPWrap::BufferSize(const FunctionCallbackInfo<Value>& args) {
 
   CHECK(args[0]->IsUint32());
   CHECK(args[1]->IsUint32());
-  int size = static_cast<int>(args[0].As<Uint32>()->Value());
 
-  if (size != args[0].As<Uint32>()->Value()) {
+  if (args[0].As<Uint32>()->Value() > std::numeric_limits<int>::max()) {
```

Comment:
Thanks!

There is a `IsInt32()` function, https://cs.chromium.org/chromium/src/v8/src/api.cc?q=IsUint32&sq=package:chromium&l=3694. 

If you want, we could replace the `>` check with `IsInt32()`. But I'm also fine with landing as is, as it's correct and gets rid of the compiler warning.

Reply:
There is a bit too much typecasting to my taste.  I'd probably assign it to a `uint32_t`, do the range check and only then assign it to an int.  `IsInt32()` would work too since the `CHECK(args[0]->IsUint32())` a few lines up establishes that it's >= 0.

Aside: the second argument is really just a bool in disguise.  The logic in lib/dgram.js could be simplified to `self._handle.bufferSize(size, buffer === 'recv')` if the code here was updated to expect a bool.

Aside aside: since this is new code, it should have been written to use the `Value()` overloads that take a `Local<Context>`.  Oh well.

Reply:
@fhinkel Nice, I was not aware of that. I'll update the PR. Thanks!

@bnoordhuis I'd be happy to take a closer look at this, but perhaps as a separate PR?